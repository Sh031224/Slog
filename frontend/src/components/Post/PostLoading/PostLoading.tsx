import React from "react";
import "./PostLoading.scss";

const PostLoading = () => {
  return (
    <div className="post-loading">
      <div className="post-loading-title" />
      <div className="post-loading-date" />
      <div className="post-loading-img" />
      <div className="post-loading-content post-loading-content-1" />
      <div className="post-loading-content post-loading-content-2" />
    </div>
  );
};

export default React.memo(PostLoading);
