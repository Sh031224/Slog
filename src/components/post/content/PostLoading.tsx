import React from "react";
import styled from "styled-components";

const PostLoading: React.FC = () => {
  return (
    <PostLoadingWrapper>
      <PostLoadingTitle />
      <PostLoadingDate />
      <PostLoadingImg />
      <PostLoadingContent width="100%" />
      <PostLoadingContent width="60%" />
    </PostLoadingWrapper>
  );
};

const PostLoadingWrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
`;

const PostLoadingTitle = styled.div`
  margin-top: 5rem;
  animation: loading 3s infinite ease-in-out;
  width: 100%;
  height: 3.4rem;
  border-radius: 5px;
  ${({ theme }) => theme.device.smallDesktop} {
    margin-top: 1rem;
    height: 3rem;
  }
`;

const PostLoadingDate = styled.div`
  width: 10rem;
  border-radius: 5px;
  margin-top: 1rem;
  animation: loading 3s infinite ease-in-out;
  height: 2rem;
`;

const PostLoadingImg = styled.div`
  width: 100%;
  height: 25rem;
  animation: loading 3s infinite ease-in-out;
  border-radius: 5px;
  margin-top: 5rem;
  ${({ theme }) => theme.device.smallDesktop} {
    height: 20rem;
    margin-top: 3rem;
  }
  ${({ theme }) => theme.device.mobile} {
    height: 18rem;
  }
`;

const PostLoadingContent = styled.div<{ width: string }>`
  margin-top: 1rem;
  height: 3.4rem;
  border-radius: 5px;
  width: ${({ width }) => width};
  animation: loading 3s infinite ease-in-out;
  ${({ theme }) => theme.device.smallDesktop} {
    height: 3rem;
  }
`;

export default PostLoading;
