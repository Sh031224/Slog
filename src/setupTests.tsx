import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import { lightTheme } from "style/theme";
import { store } from "stores";
import { Provider } from "react-redux";
import { NotificationContainer } from "react-notifications";

const Providers = ({ children }) => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={lightTheme}>
        <NotificationContainer />
        {children}
      </ThemeProvider>
    </Provider>
  );
};

const customRender = (ui, options = {}) => render(ui, { wrapper: Providers, ...options });

// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

(window as any).scrollTo = () => {};

export { customRender as render };
