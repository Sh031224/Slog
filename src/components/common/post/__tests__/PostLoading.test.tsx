import { render } from "setupTests";
import PostLoading from "../PostLoading";

describe("PostLoading", () => {
  const setup = (props = {}) => {
    return render(<PostLoading {...props} />);
  };

  it("matches snapshot", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
