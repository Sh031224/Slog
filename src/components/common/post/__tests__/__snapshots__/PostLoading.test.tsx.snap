// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostLoading matches snapshot 1`] = `
.c0 {
  width: 19rem;
  box-shadow: rgba(0,0,0,0.096) 0px 2px 10px 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #ffff;
  border-radius: 5px;
  -webkit-transition: box-shadow 0.25s ease-in 0s,-webkit-transform 0.25s ease-in 0s;
  -webkit-transition: box-shadow 0.25s ease-in 0s,transform 0.25s ease-in 0s;
  transition: box-shadow 0.25s ease-in 0s,transform 0.25s ease-in 0s;
  margin: 1rem;
  overflow: hidden;
  position: relative;
  border-radius: 5px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  position: relative;
  padding-top: 52%;
}

.c2 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-animation: loading 3s infinite ease-in-out;
  animation: loading 3s infinite ease-in-out;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
  width: 100%;
  position: relative;
}

.c4 {
  width: 80%;
  height: 1.3rem;
  -webkit-animation: loading 3s infinite ease-in-out;
  animation: loading 3s infinite ease-in-out;
  border-radius: 5px;
  margin-bottom: 0.5rem;
}

.c5 {
  height: 1rem;
  width: 100%;
  margin-top: 0.5rem;
  -webkit-animation: loading 3s infinite ease-in-out;
  animation: loading 3s infinite ease-in-out;
  border-radius: 3px;
  width: 80%;
}

.c6 {
  height: 1rem;
  width: 100%;
  margin-top: 0.5rem;
  -webkit-animation: loading 3s infinite ease-in-out;
  animation: loading 3s infinite ease-in-out;
  border-radius: 3px;
  width: 60%;
}

@media only screen and (max-width:1250px) {
  .c0 {
    width: calc(50% - 2rem);
  }
}

@media only screen and (max-width:767px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <div
    class="notification-container notification-container-empty"
  >
    <div />
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      />
      <div
        class="c5"
        width="80%"
      />
      <div
        class="c6"
        width="60%"
      />
    </div>
  </div>
</div>
`;
